### PR TITLE
[test]: add unit tests for xpath utils & tree format utils

### PR DESCRIPTION
**Pull request by seanmcguire12**
# why
- to add tests for xpath utilities and tree formatting utilities
# what changed
  - added `tests/snapshot-xpath-utils.test.ts` to cover all pure xpath helpers (`prefixXPath()`, `normalizeXPath()`, `relativizeXPath()`, `buildChildXPathSegments()`, `joinXPath()`)
  - added `tests/snapshot-tree-format-utils.test.ts` which tests formatting helpers (`formatTreeLine()`, `injectSubtrees()`, `indentBlock()`, `diffCombinedTrees()`, `cleanText()`, `normaliseSpaces()`) 
# test plan
- this is it

---
## Summary by cubic
Add unit tests for XPath utilities and tree formatting utilities to validate edge cases and prevent regressions. Tests cover path prefixing/normalization/joining, relative path derivation, positional selector building, subtree injection, diffing, indentation, and whitespace cleanup.

<sup>Written for commit df6ee849149ce9aa30c4c9bc90a554488ddbd350. Summary will update automatically on new commits.</sup>

### CODE REVIEW

File: packages/core/tests/snapshot-tree-format-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+});
+
+describe("indentBlock", () => {
+  it("prefixes each line with the provided indent", () => {
+    expect(indentBlock("a\nb", "  ")).toBe("  a\n  b");
+    expect(indentBlock("", "  ")).toBe("");
+  });
+});
+
+describe("diffCombinedTrees", () => {
```
Comment (monadoid) [inline packages/core/tests/snapshot-tree-format-utils.test.ts:91]: non-crucial but could add a test for checking that the trimming works (i.e. if prevtree has different indentation vs nextree) since it's got a bunch of logic for that?

File: packages/core/tests/snapshot-tree-format-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      ],
+    });
+
+    expect(outline).toBe(
+      "[frame-1] section: Container\n  [ax-2] button: Submit",
+    );
+  });
+});
+
+describe("injectSubtrees", () => {
```
Comment (monadoid) [inline packages/core/tests/snapshot-tree-format-utils.test.ts:33]: Could add a test for what happens when the same id appears twice, and if no ids found in the map I think?

File: packages/core/tests/snapshot-xpath-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  buildChildXPathSegments,
+  joinXPath,
+  normalizeXPath,
+  prefixXPath,
+} from "../lib/v3/understudy/a11y/snapshot/xpathUtils";
+import { relativizeXPath } from "../lib/v3/understudy/a11y/snapshot/domTree";
+
+describe("prefixXPath", () => {
+  it("treats root prefixes as no-op", () => {
+    expect(prefixXPath("/", "/div[1]")).toBe("/div[1]");
```
Comment (monadoid) [inline packages/core/tests/snapshot-xpath-utils.test.ts:13]: just fyi, you can use http://vitest.dev/api/#test-for this syntax in the future if you like it - is nice for parameterized testing - the syntax you currently have isn't too long so it doesn't matter, but it can be nice when there's more setup / LOC for each expect

**Review by monadoid — APPROVED** [review]
